add DinD feature to devcontainer.json

gregwdata opened the pull request:
Saw your tweet about getting it running in Docker locally, and thought: Why not CodeSpaces?

This PR adds the docker-in-docker feature to the codespace (docker-in-docker since CodeSpaces itself runs in a container). It is NOT yet in a finished working state.

As-is, on Codespaces, the container successfully builds, pipeline runs, and Superset starts up. But it throws an error when you open the dashboard:
![image](https://user-images.githubusercontent.com/79663385/204542021-5652c5ae-c9bb-46dc-a141-1bd0b7821fc3.png)

It looks like there's some alignment needed between where the db file gets saved inside the container (`/usr/src/app/data/data_catalog/mdsbox.db`) and where Superset expects to find it (`/workspaces/nba-monte-carlo/data/data_catalog/mdsbox.db`).

Is that already fixed in your local branch? I can proceed to get that sorted on my branch if it helps.

gregwdata committed:
add DinD feature to devcontainer.json

gregwdata committed:
change docker workdir and meltano docker env paths

gregwdata commented:
This latest commit fixes the db file location by making the dockerfile working directory, and the `meltano.yml` paths defined for the docker environment both point to `/workspaces/nba-monte-carlo`.

Everything works in codespaces now:
![image](https://user-images.githubusercontent.com/79663385/204562226-083500d0-5c04-429b-8c5c-8af127524002.png)

matsonj commented:
oh nice! I was actually working on this last night and I wasn't sure on the solution, I was actually going to have a separate ```datasources.yml``` file to load for running it inside docker, but your solution is better.

matsonj commented:
Can you add steps for running docker-in-docker in the readme?

gregwdata committed:
Add Codespaces section to readme.md

Also updated the docker-run make command in the readme to reflect recent updates.

gregwdata commented:
done. added a whole Codespaces section, since there wasn't one. No magic required to run docker-in-docker with the updated container config - the docker-in-docker plugin makes docker command line calls "just work" inside the Codespace, at least as far as my limited testing so far. (requires rebuild of any existing Codespaces you may have up though)

matsonj committed:
Update README.md

Reorganizing to move codespaces to the top.

matsonj commented:
Amazing, just reorganized this slightly.
